Runtime/LLMCharacter.cs

```diff
@@ -634,46 +586,26 @@ protected async Task<string> Slot(string filepath, string action)
         }
 
         /// <summary>
-        /// Saves the chat history and cache to the provided filename / relative path.
+        /// Saves the cache to the provided filename / relative path.
         /// </summary>
-        /// <param name="filename">filename / relative path to save the chat history</param>
+        /// <param name="filename">filename / relative path to save the cache</param>
         /// <returns></returns>
-        public virtual async Task<string> Save(string filename)
+        public virtual async Task<string> SaveCache()
         {
-            string filepath = GetJsonSavePath(filename);
-            string dirname = Path.GetDirectoryName(filepath);
-            if (!Directory.Exists(dirname)) Directory.CreateDirectory(dirname);
-            string json = JsonUtility.ToJson(new ChatListWrapper { chat = chat.GetRange(1, chat.Count - 1) });
-            File.WriteAllText(filepath, json);
-
-            string cachepath = GetCacheSavePath(filename);
             if (remote || !saveCache) return null;
-            string result = await Slot(cachepath, "save");
+            string result = await Slot(GetCacheSavePath(), "save");
```

Comment:
if we change the chathistory how can we make sure that the cache corresponds to it?

Reply:
Updated so that when the chat history is changed out, the cache will become invalidated.